witchcraft-server-config/src/runtime/de.rs

```diff
@@ -28,6 +28,14 @@ pub struct RuntimeConfig {
 #[serde(rename_all = "kebab-case")]
 pub struct DiagnosticsConfig {
     pub debug_shared_secret: String,
+    pub jemalloc: Option<super::JemallocConfig>,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct JemallocConfig {
+    pub prof_active: Option<bool>,
+    pub lg_prof_sample: Option<usize>,
```

Comment:
I opted to name these after the raw jemalloc ctl names, but I'd be open to changing them to be something more human-readable.